pymongo/uri_parser.py

```diff
@@ -1,21 +1,624 @@
-# Copyright 2024-present MongoDB, Inc.
+# Copyright 2011-present MongoDB, Inc.
 #
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# Licensed under the Apache License, Version 2.0 (the "License"); you
+# may not use this file except in compliance with the License.  You
+# may obtain a copy of the License at
 #
 # http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing, software
 # distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.  See the License for the specific language governing
+# permissions and limitations under the License.
 
-"""Re-import of synchronous URIParser API for compatibility."""
+
+"""Tools to parse and validate a MongoDB URI."""
 from __future__ import annotations
 
-from pymongo.synchronous.uri_parser import *  # noqa: F403
-from pymongo.synchronous.uri_parser import __doc__ as original_doc
+import re
+import sys
+import warnings
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Sized,
+    Union,
+    cast,
+)
+from urllib.parse import unquote_plus
+
+from pymongo.client_options import _parse_ssl_options
+from pymongo.common import (
+    INTERNAL_URI_OPTION_NAME_MAP,
+    SRV_SERVICE_NAME,
+    URI_OPTIONS_DEPRECATION_MAP,
+    _CaseInsensitiveDictionary,
+    get_validated_options,
+)
+from pymongo.errors import ConfigurationError, InvalidURI
+from pymongo.srv_resolver import _have_dnspython, _SrvResolver
+from pymongo.typings import _Address
+
+if TYPE_CHECKING:
+    from pymongo.pyopenssl_context import SSLContext
+
+_IS_SYNC = False
```

Comment:
is _IS_SYNC needed for a file like this?

Also, I think the file is synchronous.

Reply:
Good catch! Those are all artifacts from moving these files out of `pymongo/asynchronous`, fixed.